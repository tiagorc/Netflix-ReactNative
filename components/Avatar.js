import React from "react";
import { View } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import styled, { css } from "styled-components/native";

const Container = styled.TouchableOpacity`
  flex-direction: column;
  align-items: center;
`;

const avatarSize = css`
  width: ${(props) => {
    return props.big ? "100px" : "60px";
  }};
  height: ${(props) => (props.big ? "100px" : "60px")};
  margin: 10px;
`;

const AvatarIcon = styled.Image`
  ${avatarSize}
`;

const AvatarText = styled.Text`
  color: white;
`;

const Overlay = styled.View`
  ${avatarSize}
  position: absolute;
  background-color: black;
  opacity: 0.8;
  justify-content: center;
  align-items: center;
`;

const OverlayContainer = styled.View`
  ${avatarSize}
  position: absolute;
  justify-content: center;
  align-items: center;
`;

const Editable = (props) => {
  return props.edit ? (
    <OverlayContainer big={props.big}>
      <Overlay big={props.big} />
      <MaterialIcons name="edit" size={24} color="white" />
    </OverlayContainer>
  ) : null;
};

const Avatar = (props) => {
  return (
    <Container style={{ alignItems: "center" }} onPress={props.onPress}>
      {props.uri ? (
        <View>
          <AvatarIcon
            big={props.big}
            source={{ isStatic: true, uri: props.uri }}
          />
          <Editable edit={props.edit} {...props} />
        </View>
      ) : (
        <View>
          <AvatarIcon big={props.big} source={props.image} />
          <Editable edit={props.edit} {...props} />
        </View>
      )}

      <AvatarText>{props.name}</AvatarText>
    </Container>
  );
};

export default Avatar;
