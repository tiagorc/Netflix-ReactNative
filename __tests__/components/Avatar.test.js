import React from "react";
import Avatar from "../../components/Avatar";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Avatar />).toJSON();
  expect(tree).toMatchSnapshot();
});
