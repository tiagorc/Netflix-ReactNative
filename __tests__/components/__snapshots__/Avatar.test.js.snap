// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "opacity": 1,
    }
  }
>
  <View>
    <Image
      style={
        Array [
          Object {
            "height": 60,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": 60,
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
      ]
    }
  />
</View>
`;
